segmentation_labeling_app/rois/rois.py

```diff
@@ -0,0 +1,170 @@
+
+        shape = query_utils.query(
+            f"SELECT * FROM public.segmentation_runs WHERE id={segmentation_id}",
+            user=label_vars.user,
+            host=label_vars.host,
+            database=label_vars.database,
+            port=label_vars.port,
+            password=label_vars.password)[0]['video_shape']
+
+        roi = query_utils.query(
+            f"SELECT * FROM public.rois WHERE "
+            f"segmentation_run_id={segmentation_id} AND id={roi_id}",
+            user=label_vars.user,
+            host=label_vars.host,
+            database=label_vars.database,
+            port=label_vars.port,
+            password=label_vars.password)
+        return ROI(coo_rows=roi['coo_rows'][0],
+                   coo_cols=roi['coo_cols'][0],
+                   coo_data=roi['coo_data'][0],
+                   image_shape=shape,
+                   segmentation_id=segmentation_id,
+                   roi_id=roi_id)
+
+    def generate_binary_mask_from_threshold(self, threshold: float):
+        """
+        Simple binary mask from a provided threshold.
+        Args:
+            threshold: The threshold to compare values
+
+        Returns:
+
+        Notes:
+            This function will likely become deprecated as we work on more
+            applicable binary thresh holding
+        """
+        return np.where(self._sparse_coo.toarray() >= threshold, 1, 0)
+
+    def generate_roi_inner_edge_coordinates(self, stroke_size: int,
+                                            threshold: float):
+        """
+        Returns a list of coordinates in a video frame for the inner outline
+        of an ROI given a desired stroke size. Looks at each border pixel and
+        moves away from edge by stroke specified pixels. This process is completed
+        by defining the center of the roi and creating a unit vector from the
+        center to the edge point. This unit vector is then multiplied by the stroke
+        and a slight weight to correct for non int centers. The movement vector
+        is added to the edge point and the new point is then set to true in
+        the return mask.
+        Args:
+            stroke_size: Value in pixels for the size of the stroke
+            threshold: Float to threshold every pixel against to generate
+            binary mask
+
+        Returns:
+            eroded_inner_mask: A binary mask for the new edge coordinates
+            that has been slightly eroded
+            coordinate_list: A list of coordinates for the edge of the ROI
+            pushed in by stroke pixel count
+        """
+        edge_coordinates = self.get_edge_points(threshold=threshold)
+        center = self.get_centroid_of_thresholded_mask(threshold=threshold)
+        inner_edge_mask = np.zeros(shape=self.image_shape)
+        for edge_coordinate in edge_coordinates[0]:
+            # get distance to center
+            vector = np.array([(center[0] - edge_coordinate[0][0]),
```

Comment:
I think this whole function can probably be out-sourced to some package, but, fwiw, you should check out:
https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance.cdist.html
this would probably help you vectorize this loop, making it faster and easier to read.